compiler/src/dmd/errors.d

```diff
@@ -819,8 +819,11 @@ private void colorHighlightCode(ref OutBuffer buf)
     }
     ++nested;
 
-    auto gaggedErrorsSave = global.startGagging();
-    scope Lexer lex = new Lexer(null, cast(char*)buf[].ptr, 0, buf.length - 1, 0, 1, global.errorSink, global.vendor, global.versionNumber());
+    __gshared ErrorSinkNull errorSinkNull;
+    if (!errorSinkNull)
```

Comment:
This condition is not necessary. 

```d
gshared ErrorSinkNull errorSinkNull = new ErrorSinkNull();
```

Should suffice, as `__gshared` variables are initialized at compile time.